src/shared/contexts/pinneditems.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {
+  createContext,
+  FC,
+  useEffect,
+  useState,
+  memo,
+  useCallback,
+  ReactElement,
+} from 'react'
+import {CLOUD} from '../constants'
+
+let getPinned
+let putPinned
+let deletePinned
+let postPinned
+
+if (CLOUD) {
+  const pinnedItemFunctions = require('src/client/pinnedItemRoutes')
+  getPinned = pinnedItemFunctions.getPinned
+  putPinned = pinnedItemFunctions.putPinned
+  postPinned = pinnedItemFunctions.postPinned
+  deletePinned = pinnedItemFunctions.deletePinned
+}
+
+export interface PinnedItem {
+  orgID: string
+  userID: string
+  id: string
+  createdAt: string
+  updatedAt: string
+  metadata: {
+    [key: string]: string
+  }
+  type: string
+}
+
+export enum PinnedItemTypes {
+  Dashboard = 'dashboard',
+  Cell = 'cell',
+  Notebook = 'notebook',
+  Task = 'task',
+  Alert = 'alert',
+  UploadCSV = 'uploadCsv',
+  UploadLP = 'uploadLP',
+  ClientLibrary = 'clientLibrary',
+}
+export const PinnedItemsContext = createContext(null)
+
+export const getPinnedItems = async () => {
+  const pinnedRes = await getPinned({})
+  if (pinnedRes.status !== 200) {
+    throw new Error('Unable to retrieve pinned items')
+  } else {
+    return pinnedRes.data
+  }
+}
+
+export const addPinnedItem = async (item: Partial<PinnedItem>) => {
+  const pinnedRes = await postPinned({data: item})
+  if (pinnedRes.status !== 200) {
+    throw new Error('Unable to retrieve pinned items')
+  }
+}
+
+export const deletePinnedItemByParam = async (param: string) => {
+  try {
+    const items = await getPinnedItems()
```

Comment:
pretty sure this will need an empty `{}`
```suggestion
    const items = await getPinnedItems({})
```

Reply:
nope, bc its the wrapper